core/src/main/java/ai/timefold/solver/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -1582,9 +1583,19 @@ <C> TriConstraintStream<A, B, C> concat(@NonNull TriConstraintStream<A, B, C> ot
             @NonNull Function<A, B> paddingFunction) {
         var firstStream = this;
         var remapped = firstStream.map(ConstantLambdaUtils.biPickFirst());
-        var secondStream = getConstraintFactory().forEach(otherClass)
-                .ifNotExists(remapped, Joiners.equal());
-        return firstStream.concat(secondStream, paddingFunction);
+
+        if (firstStream instanceof AbstractConstraintStream<?> abstractConstraintStream) {
```

Comment:
When would a `BiConstraintStream` not be an instance of `AbstractConstraintStream`? What else would it be an instance of?